mobile/app/(app)/change-password/index.tsx

```diff
@@ -0,0 +1,205 @@
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import API from "../../../services/api";
+import ChangePasswordConfirmation from "./change-password-confirmation";
+
+interface FormData {
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+}
+
+const ChangePassowrd = () => {
+  const insets = useSafeAreaInsets();
+  const { t } = useTranslation("change_password");
+  const [credentialsError, setCredentialsError] = React.useState(false);
+  const [succesfullyChanged, setSuccesfullyChanged] = React.useState(false);
+
+  // Form validation schema
+  const formSchema = z
+    .object({
+      currentPassword: z.string().min(1, t("form.current_password.required")),
+      newPassword: z
+        .string()
+        .min(8, t("form.new_password.helper"))
+        .regex(/^(?=.*[A-Z])(?=.*[0-9])(?=.*[^a-zA-Z0-9]).*$/, t("form.new_password.pattern")),
+      confirmPassword: z.string(),
+    })
+    .refine((data) => data.newPassword === data.confirmPassword, {
+      message: t("form.confirm_password.no_match"),
+      path: ["confirmPassword"],
+    });
+
+  // React Hook Form setup
+  const { handleSubmit, control, formState } = useForm<FormData>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      currentPassword: "Votemonitor1*",
+      newPassword: "Votemonitor1*",
+      confirmPassword: "Votemonitor1*",
+    },
+  });
+  const { errors } = formState;
+
+  // Submit handler - change password
+  const onSubmit = async (data: FormData) => {
+    try {
+      await API.post("auth/change-password", {
+        password: data.currentPassword,
+        newPassword: data.newPassword,
+        confirmNewPassword: data.confirmPassword,
+      });
+      setSuccesfullyChanged(true);
+    } catch (err) {
+      console.log("Error while changing password...", err);
+      setCredentialsError(true);
+    }
+  };
+
+  // Render either form or confirmation screen
+  return succesfullyChanged ? (
+    <ChangePasswordConfirmation />
+  ) : (
+    <Screen
+      preset="auto"
+      backgroundColor="white"
+      ScrollViewProps={{
+        bounces: false,
+      }}
+      contentContainerStyle={$containerStyle}
+    >
+      <Header
+        title={t("header.title")}
+        titleColor="white"
+        barStyle="light-content"
+        leftIcon={<Icon icon="chevronLeft" color="white" />}
+        onLeftPress={() => router.back()}
+      />
+
+      <Form control={control} errors={errors} credentialsError={credentialsError}></Form>
+
+      <Card width="100%" paddingBottom={16 + insets.bottom} marginTop="auto">
+        <Button onPress={handleSubmit(onSubmit)}>{t("form.actions.save_password")}</Button>
+      </Card>
+    </Screen>
+  );
+};
+
+const Form = ({
+  control,
+  errors,
+  credentialsError,
+}: {
+  control: Control<FormData, any>;
+  errors: FieldErrors<FieldValues>;
+  credentialsError: boolean;
+}) => {
+  const { t } = useTranslation("change_password");
+
+  return (
```

Comment:
Is there a point why is this in a separate component and it's not part of the page?

Reply:
Done, now is part of the page.